proposals/11385-literal-input-parameters/README.md

```diff
@@ -0,0 +1,138 @@
+
+### Goals
+The goal of this KEP is to expand supported pipeline input to include `typing.Literal`:
+- A pipeline author can write pipelines and pipeline components with input Literal parameters, and the SDK compiler can parse Literal parameters and output a pipeline YAML containing the Literal elements.
+- When a user submits a pipeline YAML file with literal inputs and its corresponding runtime values to the API server, the pipeline run server validates the input before submitting the run. If the input is not present in the Literal parameter, the pipeline run server rejects the request and returns an error.
+- When a user runs a pipeline with Literal input via the UI, they should be able to select a valid input from a drop-down menu in the runtime parameter box. 
+
+### Non-Goals
+This KEP is not proposing the creation of an additional custom data type for Literal parameters.
+
+## Proposal
+### User Stories
+#### Story #1:
+I am a machine learning engineer writing a pipeline with components that utilize GPU accelerators, and I have two specific options that can be used: `nvidia-tesla-k80` or `nvidia-tesla-p100`. If any other option is input, the component fails. Even if I set component-level input validation, the pipeline still will not fail until this component executes. This is the 10th component in the pipeline, and this failure wastes a lot of time and resources. I want to restrict my input to `nvidia-tesla-k80` and `nvidia-tesla-p100`.
+
+#### Story #2:
+I am a pipeline author concerned about malicious users inputting values that could cause my pipeline to run malicious code. I want to limit the input to a specific set of predetermined values to prevent this.
+
+### Design Details
+In order to streamline the design and also to account for there being no `typing.Literal` counterpart in Go, a Literal parameter is represented similarly to a more typical parameter (e.g. string, boolean) with the only difference being an additional `literals` field:
+#### InputSpec
+`literals: Optional[List[Any]]` is added to the InputSpec class definition in `kfp/dsl/structures.py`:
+```aiignore
+class InputSpec:
+    type: Union[str, dict]
+    default: Optional[Any] = None
+    literals: Optional[List[str] | List[int] | List[float]] = None    optional: bool = False
+    is_artifact_list: bool = False
+    description: Optional[str] = None
+```
+For example, a pipeline component with a Literal parameter would look like this:
+```aiignore
+@dsl.component()
+def component(input: Literal["a", "b", "c"]):
```

Comment:
Could you describe the behavior of literal inputs that are optional and/or have a default?